Organ storage bag can insert organs into a smart organ storage

Sarchutar opened the pull request:
## About The Pull Request

Makes organ storage bags (from medical cyborgs) able to insert the organs they hold into a smart organ storage (organ smartfridge).

## Why It's Good For The Game

Medical cyborgs currently have no way to insert organs into the smart organ storage, this changes that.

## Changelog
:cl:
add: Medical cyborgs can insert organs with the organ storage bag into a smart organ storage
/:cl:

Sarchutar committed:
initial thing

-makes organs inside the organ storage bag be insertable into the smart organ storage
-the organ storage bag checks if the thing it tries to insert into itself is an actual item

MitztheKat commented:
yeees mediborgs go brr
I have no vote power but I have the power to leave a comment and possibly raise this commit's visibility

park66665 commented on code/modules/surgery/tools.dm:
```diff
@@ -247,6 +247,8 @@
 	if(!proximity)
 		return
+	if(!isitem(I))		//Check if it is an actual item
+		return
```
I don't think this check is needed. There's an item type check just a few lines below.

Sarchutar replied:
If you mean this `if(!isorgan(I) && !isbodypart(I))` then yes, but if you click on anything other than an organ or a bodypart, like a machine or something then it will always show the message that the bag can only hold bodyparts.

park66665 replied:
1. The code comment doesn't describe your intention.
2. I am not convinced of your intention.

Therefore, you really should move this line right above the existing type check and add a comment *explicitly* stating that you want this type of action to fail without a warning. (You also probably can make it work via some intelligent usage of whitespaces. Go figure.)

Or you remove your new check. Your call.

Sarchutar replied:
I made this check because currently if you click on literally anything like walls, tiles, machines etc. the other checks _will_ trigger because either the bag is full or the thing is not an organ or a bodypart. I added the check specifically to check if the thing is an item to reduce the cases where this will happen, that is also why this was above both of the checks and not only above the `if(!isorgan(I) && !isbodypart(I))` check.

But i removed it entirely since it is not _that_ needed.

Sarchutar committed:
remove the check

-removes the check for the organ storage bag to check if it tries to insert an item or not

park66665 reviewed (approved):
All clear. This PR is good to go.
